### ISSUE TITLE
ES6 modules can be reordered incorrectly

### ISSUE THREAD

**Issue by azakus** [google/closure-compiler#2362]
I have a project which seems to make Closure Compiler reorder ES6 modules improperly, breaking functionality.

A small reproduction of this issue can be found at https://github.com/azakus/closure-reorder-import

In this reproduction, I use rollup and the npm package for closure on the same entry point. Rollup will produce the expected output, while closure seems to reorder the modules and produce an incorrect output

@ChadKillingsworth

**Comment by ChadKillingsworth** [comment]
Sorry - I've been ushering through a lot of other PRs lately. I'll try to look at this soon.

**Comment by ChadKillingsworth** [comment]
For reference, the module tree looks like:

```
entry
 ├ b/b.js
 |   └ b/c.js
 ├ b/a.js
 ├ important.js
 ├ a/b.js
 └ a/a.js
```

**Comment by ChadKillingsworth** [comment]
Closed by #2641
